qiskit/transpiler/passes/layout/vf2_post_layout.py

```diff
@@ -339,7 +341,10 @@ def run(self, dag):
                 )
                 break
         if chosen_layout is None:
-            stop_reason = VF2PostLayoutStopReason.NO_SOLUTION_FOUND
+            if initial_layout is not None:
+                stop_reason = VF2PostLayoutStopReason.NO_BETTER_SOLUTION_FOUND
+            else:
+                stop_reason = VF2PostLayoutStopReason.NO_SOLUTION_FOUND
```

Comment:
If we reach all the way down here, we must have found a solution, I think?  Either we're in the situation where our `initial_layout` was the best, or something else must have overwritten `chosen_layout`.  It feels a little like we might want to initialise `stop_reason=NOTHING_BETTER` and `chosen_layout=initial_layout` up top when we're scoring the initial layout, so we can localise any "inferences".

Reply:
I like setting `chosen_layout=initial_layout` at the top of the code and then comparing against this without setting the post_layout field - it makes the intend clearer. I don't know about setting `stop_reason=NOTHING_BETTER` as default but will look into it tomorrow. :-)

Reply:
Yea makes sense!